transform compile error in Visual Studio

I'm using `transformed` to convert a QVector<struct> into a QJsonArray of QJsonObjects as per the code below. It compiles OK with g++ 10 and clang++ 11, but not in Visual Studio 2022.

```
#include <kdalgorithms.h>

#include <QVector>
#include <QJsonArray>
#include <QJsonObject>

void f()
{
	struct S { int a, b; };
	QVector<S> s;

	auto j = kdalgorithms::transformed<QJsonArray>(s, [](const auto& v)
		{
			return QJsonObject{
				{ QStringLiteral("a"), v.a },
				{ QStringLiteral("b"), v.b }
			};
		});
}
```

The VS compiler reports

```
1>test.cpp(973,30): error C2039: 'a': is not a member of 'QJsonValue'
1>C:\qt\Qt5.15.12\5.15.12\msvc2019_64\include\QtCore\qjsonvalue.h(59,21): message : see declaration of 'QJsonValue'
1>C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.34.31933\include\type_traits(1470,1): message : see reference to function template instantiation 'auto f::<lambda_1>::operator ()<QJsonValue>(const _T1 &) const' being compiled
1>        with
1>        [
1>            _T1=QJsonValue
1>        ]
1>C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.34.31933\include\type_traits(1675,38): message : see reference to alias template instantiation 'std::_Decltype_invoke_nonzero<f::<lambda_1>,QJsonValue,>' being compiled
1>...\include\kdalgorithms\bits\transform.h(82,19): message : see reference to variable template 'const bool is_invocable_v<`f'::`2'::<lambda_1>,QJsonValue>' being compiled
1>test.cpp(974,30): error C2039: 'b': is not a member of 'QJsonValue'
1>C:\qt\Qt5.15.12\5.15.12\msvc2019_64\include\QtCore\qjsonvalue.h(59,21): message : see declaration of 'QJsonValue'
1>test.cpp(972,22): error C2440: '<function-style-cast>': cannot convert from 'initializer list' to 'QJsonObject'
1>test.cpp(972,22): message : No constructor could take the source type, or constructor overload resolution was ambiguous
```

However if I replace the `const auto& v` parameter in the lambda with `const S& v` then it does compile.

It seems that with auto a QJsonValue is being passed, being the output container's value_type perhaps?

--- a/src/bits/transform.h
+++ b/src/bits/transform.h
@@ -128,15 +128,15 @@ auto transformed_with_new_return_type(InputContainer &&input, Transform &&transf
 }
 
 // -------------------- transform --------------------
-template <typename Container, typename UnaryPredicate>
-void transform(Container &input, UnaryPredicate &&predicate)
+template <typename Container, typename Transform>
+void transform(Container &input, Transform &&predicate)
 #if __cplusplus >= 202002L
-    requires UnaryPredicateOnContainerValues<UnaryPredicate, Container>
+    requires std::is_invocable_r_v<ValueType<Container>, Transform, ValueType<Container>>
 #endif
 
 {
     std::transform(std::begin(input), std::end(input), std::begin(input),
-                   std::forward<UnaryPredicate>(predicate));
+                   std::forward<Transform>(predicate));
 }
 
 // -------------------- filtered_transformed --------------------

--- a/tests/tst_kdalgorithms.cpp
+++ b/tests/tst_kdalgorithms.cpp
@@ -781,6 +781,12 @@ void TestAlgorithms::transform()
     kdalgorithms::transform(vec, squareItem);
     std::vector<int> expected{1, 4, 9, 16};
     QCOMPARE(vec, expected);
+
+    QStringList strings{"abc", "def", "hij"};
+    auto to_upper = [](const QString &string) { return string.toUpper(); };
+    kdalgorithms::transform(strings, to_upper);
+    QStringList expected_strings{"ABC", "DEF", "HIJ"};
+    QCOMPARE(strings, expected_strings);
 }
 
 void TestAlgorithms::filtered_transformed()

